components/script/dom/htmlimageelement.rs

```diff
@@ -1032,3 +1053,228 @@ fn image_dimension_setter(element: &Element, attr: LocalName, value: u32) {
+        where F: Fn(&char) -> bool
+        {
+            for (i, ch) in s.chars().enumerate() {
+                if !predicate(&ch) {
+                    return (&s[0..i], &s[i..])
+                }
+            }
+        return (s, "");
+        }
+
+pub fn parse_a_srcset_attribute(input: String) -> Vec<ImageSource> {
+    let mut start = 0;
+    let mut candidates: Vec<ImageSource> = Vec::new();
+    loop {
+        let position = &input[start..];
+
+        let(spaces, position) = collect_sequence_characters(position, |c| *c ==',' || char::is_whitespace(*c));
+        let x = spaces.find(',');
+        match x {
+            Some(val) => println!("Parse Error"),
+            None => println!("No commas\n"),
+        }
+
+        // add the counts of spaces that we collect to advance the start index
+        let space_len = spaces.char_indices().count();
+        start += space_len;
+
+        //Returns and breaks out of the loop
+        if position.is_empty() {
+            return candidates;
+        }
+        let(url, spaces) = collect_sequence_characters(position, |c| !char::is_whitespace(*c));
+
+        // add the counts of urls that we parse to advance the start index
+        start += url.char_indices().count();
+
+        let comma_count = url.chars().rev().take_while(|c| *c == ',').count();
+        let url: String = url.chars().take(url.chars().count() - comma_count).collect();
+        if comma_count > 1 {
+            println!("Parse Error (trailing commas)")
+        }
+
+        // add 1 to start index, for the comma
+        start += 1;
+
+        let(space, position) = collect_sequence_characters(spaces, |c| char::is_whitespace(*c));
+
+        let space_len = space.char_indices().count();
+        start += space_len;
+
+        let mut descriptors = LinkedList::<String>::new();
+        let mut current_descriptor = String::new();
+        let mut state = ParseState::InDescriptor;
+        let mut char_stream = position.chars().enumerate();
+        //let mut last_index = 0;
+        let mut buffered: Option<(usize, char)> = None;
+        loop {
+            let nextChar = buffered.take().or_else(|| char_stream.next());
+            if let Some((i, _)) = nextChar {
+                start += 1;
+            }
+            match state {
+                ParseState::InDescriptor => {
+                    match nextChar {
+                        Some((idx, c @ ' ')) => {
+                            if !current_descriptor.is_empty() {
+                                descriptors.push_back(current_descriptor.clone());
+                                current_descriptor = String::new();
+                                state = ParseState::AfterDescriptor;
+                            }
+                            continue;
+                        }
+                        Some((idx, c @ ',')) => {
+                            position.chars().enumerate();
```

Comment:
What's this?

Reply:
This should be removed, since it does not do anything.